src/ser.rs

```diff
@@ -135,18 +133,32 @@ pub fn to_string<T: ?Sized>(value: &T) -> Result<String, Error>
 ///         },
 ///     };
 ///
-///     let mut file = File::create("config.toml").unwrap();
-///     toml::to_writer(&mut file, &config).unwrap();
+///     let mut writer = io::Cursor::new(Vec::default());
+///     toml::to_writer(&mut writer, &config)?;
+///
+///     assert_eq!(
+///     std::str::from_utf8(&writer.into_inner()).unwrap(),
+///     r#"[database]
+///ip = "192.168.1.1"
+///port = [8001, 8002, 8003]
+///connection_max = 5000
+///enabled = false
+///"#
+///    );
+///
+///    Ok(())
 /// }
 /// ```
-pub fn to_writer<W, T: ?Sized>(writer: &mut W, value: &T) -> Result<(), Error>
+pub fn to_writer<W, T: ?Sized>(writer: &mut W, value: &T) -> Result<(), io::Error>
```

Comment:
Why not just use `io::Result`?

Reply:
It's the same because io::Result is an alias of result::Result<T, Error>  :). If you prefer, i can put that.
https://doc.rust-lang.org/src/std/io/error.rs.html#41-42

I have let that because all the class use Result<T, E>

Reply:
Just my two cents, I am not a maintainer of this project anyway.